src/main/java/edu/usf/cutr/grha/ProcessorMain.java

```diff
@@ -16,32 +16,39 @@
 package edu.usf.cutr.grha;
 
 import edu.usf.cutr.grha.io.ChicagoTncParser;
+import edu.usf.cutr.grha.io.CsvParser;
 import edu.usf.cutr.grha.io.GPSTestExtendedHeaderParser;
 import edu.usf.cutr.grha.io.GPSTestParser;
 import edu.usf.cutr.grha.utils.IOUtils;
 
+import java.io.FileNotFoundException;
+import java.io.InputStream;
+
 public class ProcessorMain {
 
-    public static void main(String[] args) {
+    public static void main(String[] args) throws FileNotFoundException {
         if (args.length != 2) {
             System.out.println("The first command-line parameter should be the GPSTest data filename with simple headers, and the 2nd parameter should be the filename of the Chicago dataset");
             System.exit(1);
         }
 
-        GPSTestParser gpsTestParser = new GPSTestParser(args[0]);
+        InputStream inputStream0 = new CsvParser().getInputStream(args[0]);
```

Comment:
The base class `CsvParser` shouldn't be instantiated here - you're using it more like a utility method, and in that case the code should be moved to a utility class.